Source/WebCore/Modules/cookie-store/CookieStore.cpp

```diff
@@ -467,8 +467,20 @@ void CookieStore::set(CookieInit&& options, Ref<DeferredPromise>&& promise)
         }
     }
 
-    if (options.expires)
-        cookie.expires = *options.expires;
+    if (options.expires) {
+        // CFNetwork floors the creation/expiration times of NSHTTPCookies.
+        // When this cookie is converted to an NSHTTPCookie, if the creation and expiration
+        // differ by less than 1 second, flooring them may reduce the difference to 0 seconds.
+        // This can cause the onchange event to wrongly fire as a deletion when it was a change.
+        // To get around this flooring, add/subtract 1 second from the expiration.
+        auto expires = *options.expires;
+        if (expires > cookie.created)
+            expires += 1000.0;
+        else if (expires < cookie.created)
```

Comment:
If `expires < cookie.created`, isn't the cookie deleted anyways? Why do we need this hack?

Reply:
This is true, the cookie is deleted anyways. I put this here because even if `expires < cookie.created`, the difference could be reduced to 0 and I wasn't sure if CFNetwork would register that as a deletion. But I just tested without this and it does seem to register as a deletion. So I've removed this.